internal/app/init/pkg/system/services/ntpd.go

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// nolint: dupl,golint
+package services
+
+import (
+	"github.com/autonomy/talos/internal/app/init/pkg/ntpd"
+	"github.com/autonomy/talos/internal/app/init/pkg/system/conditions"
+	"github.com/autonomy/talos/internal/pkg/userdata"
+)
+
+// NTPd implements the Service interface. It serves as the concrete type with
+// the required methods.
+type NTPd struct {
+	client *ntpd.NTP
+}
+
+// ID implements the Service interface.
+func (t *NTPd) ID(data *userdata.UserData) string {
+	return "ntpd"
+}
+
+// PreFunc implements the Service interface.
+func (t *NTPd) PreFunc(data *userdata.UserData) error {
+	n := ntpd.New(data.Services.NTPd)
+	if err := n.Query(); err != nil {
+		return err
+	}
+	if err := n.SetTime(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// PostFunc implements the Service interface.
+func (t *NTPd) PostFunc(data *userdata.UserData) (err error) {
+	return nil
+}
+
+// ConditionFunc implements the Service interface.
+func (t *NTPd) ConditionFunc(data *userdata.UserData) conditions.ConditionFunc {
+	// what is this?
```

Comment:
The conditions allow us to do things like delay the start of a service until a file or files exist on disk.